src/Chapter3.hs

```diff
@@ -373,9 +378,17 @@ after the fight. The battle has the following possible outcomes:
    doesn't earn any money and keeps what they had before.
 
 -}
+data Knight = Knight Int Int Int deriving (Show)
+data Monster = Monster Int Int Int deriving (Show)
```

Comment:
This is a correct definition. However, for readability and maintainability, It's better to use records here. Otherwise, people see 3 values of type `Int` but it's not clear what each `Int` represent. Moreover, after using records, you will be able to define the `fight` function easily without pattern matching.